### PR TITLE
Docker node red integration

### PR DESCRIPTION
**Pull request by callemand**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)

### Description of change
Provide a new service to start node-red into docker and follow the status.

Screenshot 2023-06-08 at 15 11 38
Screenshot 2023-06-08 at 15 11 52

### CODE REVIEW

File: front/src/routes/integration/all/node-red/NodeRedPage.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const NodeRedPage = ({ children, user }) => (
 
                   <DeviceConfigurationLink
                     user={user}
-                    documentKey="nodeRed"
+                    documentKey="noder-red"
```
Comment (Pierre-Gilles) [inline front/src/routes/integration/all/node-red/NodeRedPage.js]: little typo, "noder-red", it's "node-red"

File: front/src/config/i18n/en.json [hunk 2]
```diff
@@ -598,6 +598,41 @@
       "saveError": "There was an error saving the device.",
       "documentation": "Zigbee2mqtt documentation"
     },
+    "nodeRed": {
+      "title": "Node-RED",
+      "description": "\"Control your devices with Node-RED.",
```
Comment (Pierre-Gilles) [inline front/src/config/i18n/en.json]: Why the "\"

File: server/lib/system/system.getGladysBasePath.js [hunk 3]
```diff
@@ -11,16 +13,20 @@ async function getGladysBasePath() {
...
+    if (gladysMounts) {
+      const baseMount = gladysMounts.find((mount) => {
+        return mount.Destination === basePathOnContainer;
+      });
+      if (baseMount) {
+        return { basePathOnContainer, basePathOnHost: baseMount.Source };
+      }
     }
+  } catch (e) {
+    logger.warn(`Node-RED: Error while fetching container mounts: ${e.message}`);
```
Comment (Pierre-Gilles) [inline server/lib/system/system.getGladysBasePath.js]: You are modifying a system function with Node-RED specific code

File: server/services/node-red/package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  ],
+  "cpu": [
+    "x64",
+    "arm",
+    "arm64"
+  ],
+  "dependencies": {
+    "bcrypt": "^5.1.0",
+    "fs-extra": "^11.1.1",
+    "jszip": "^3.10.1",
```
Comment (Pierre-Gilles) [inline server/services/node-red/package.json]: Where is it used?

File: server/services/node-red/package.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    "lodash.clonedeep": "^4.5.0",
+    "mqtt": "^4.2.6"
```
Comment (Pierre-Gilles) [inline server/services/node-red/package.json]: Where is it used?

File: server/services/node-red/package.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    "darwin",
+    "linux",
+    "win32"
+  ],
+  "cpu": [
+    "x64",
+    "arm",
+    "arm64"
+  ],
+  "dependencies": {
```
Comment (Pierre-Gilles) [inline server/services/node-red/package.json:15]: Check all deps, seems to be a duplicate

Re-run npm install after to update package-lock.json

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thanks for this PR ! 🙏 

A few feedbacks

File: front/src/config/i18n/en.json (hunk 2 above)
Comment (callemand) [reply front/src/config/i18n/en.json]: Mistake

File: server/lib/system/system.getGladysBasePath.js [hunk 7]
```diff
@@ -11,16 +13,20 @@ async function getGladysBasePath() {
...
+    const gladysMounts = await this.getContainerMounts(currentContainerId);
+    if (gladysMounts) {
+      const baseMount = gladysMounts.find((mount) => {
+        return mount.Destination === basePathOnContainer;
+      });
+      if (baseMount) {
+        return { basePathOnContainer, basePathOnHost: baseMount.Source };
+      }
     }
+  } catch (e) {
```
Comment (Pierre-Gilles) [inline server/lib/system/system.getGladysBasePath.js]: I'm not sure we want to change this system API in this PR. Other integrations relies on this function already

File: server/test/lib/system/system.getGladysBasePath.test.js [hunk 8]
```diff
@@ -49,6 +49,14 @@ describe('system.getGladysBasePath', () => {
       basePathOnContainer: '/var/lib/gladysassistant',
     });
   });
+  it('should return default basePath because failed to get mount', async () => {
```
Comment (Pierre-Gilles) [inline server/test/lib/system/system.getGladysBasePath.test.js]: Same, not sure we want to soft fail for that, in this PR

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]

File: server/lib/system/system.getGladysBasePath.js (hunk 7 above)
Comment (callemand) [reply server/lib/system/system.getGladysBasePath.js]: Ok, I have removed this change

File: server/test/lib/system/system.getGladysBasePath.test.js (hunk 8 above)
Comment (callemand) [reply server/test/lib/system/system.getGladysBasePath.test.js]: Ok, I have removed this change

**Review by Pierre-Gilles — APPROVED** [review]
Great PR, thanks for your work 🙏 

Good for me!

**Pull request merged**